python/pyspark/serializers.py

```diff
@@ -62,11 +62,12 @@
 if sys.version < '3':
     import cPickle as pickle
     from itertools import izip as zip, imap as map
+    pickle_protocol = 2
 else:
     import pickle
     basestring = unicode = str
     xrange = range
-pickle_protocol = pickle.HIGHEST_PROTOCOL
+    pickle_protocol = 3
```

Comment:
We can use `pickle.DEFAULT_PROTOCOL` too but let me stick with constant since seems protocol 4 has this bug.

Reply:
It looks nice and solid.

Reply:
Do you mean the bug issue related to the root cause somewhere? (I think) it's more like an issue within Pryolite library .. I am not 100% sure yet. I will update that when I have it. I am looking into this to identify the cause.